src/main/java/com/nitnelave/CreeperHeal/block/CreeperMultiblock.java

```diff
@@ -0,0 +1,88 @@
+package com.nitnelave.CreeperHeal.block;
+
+import com.nitnelave.CreeperHeal.config.CfgVal;
+import com.nitnelave.CreeperHeal.config.CreeperConfig;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Abstract CreeperBlock representing an item that may span multiple blocks.
+ *
+ * @author Jikoo
+ */
+public abstract class CreeperMultiblock extends CreeperBlock
+{
+
+    final Set<BlockState> dependents;
+
+    CreeperMultiblock(BlockState blockState)
+    {
+        super(blockState);
+        this.dependents = new HashSet<BlockState>();
+    }
+
+    @Override
+    public void update()
+    {
+        super.update();
+        for (BlockState dependent : dependents)
+        {
+            dependent.update(true, false);
+        }
+    }
+
+    @Override
+    protected boolean checkForDrop()
+    {
+        if (checkForDropHelper(getBlock()))
+        {
+            return true;
+        }
+
+        for (BlockState dependent : dependents)
+        {
+            if (checkForDropHelper(dependent.getBlock()))
+            {
+                return true;
+            }
+        }
+
+        return false;
+
+    }
+
+    private boolean checkForDropHelper(Block block)
```

Comment:
Some of this ends up being redone later in my fix for the dupe bugs related to incorrect drops that's been present since 2013. I'll check if it's worth changing, not sure how much even survives in that patch.

Reply:
After reviewing the changes involved I'd like to postpone this until I create a PR backporting f3b468b to 1.11, which will be next.